models/location_helpers.go

```diff
@@ -138,51 +211,57 @@ func CreateBackupLocation(q *reform.Querier, params CreateBackupLocationParams)
 		Name:        params.Name,
 		Description: params.Description,
 	}
-	configCount := 0
-	if params.S3Config != nil {
-		configCount++
-	}
-	if params.PMMServerConfig != nil {
-		configCount++
-	}
-	if params.PMMClientConfig != nil {
-		configCount++
+
+	params.FillLocation(row)
+
+	if err := q.Insert(row); err != nil {
+		return nil, errors.Wrap(err, "failed to create backup location")
 	}
 
-	if configCount > 1 {
-		return nil, status.Error(codes.InvalidArgument, "Only one config is allowed.")
+	return row, nil
+}
+
+// ChangeBackupLocationParams are params for updating existing backup location.
+type ChangeBackupLocationParams struct {
+	Name        string
+	Description string
+
+	BackupLocationsConfigs
+}
+
+// ChangeBackupLocation updates existing location by specified locationID and params.
+func ChangeBackupLocation(q *reform.Querier, locationID string, params ChangeBackupLocationParams) (*BackupLocation, error) {
+	if err := params.Validate(); err != nil {
+		return nil, err
 	}
 
-	switch {
-	case params.S3Config != nil:
-		if err := checkS3Config(params.S3Config); err != nil {
-			return nil, err
-		}
-		row.Type = S3BackupLocationType
-		row.S3Config = params.S3Config
+	row, err := FindBackupLocationByID(q, locationID)
+	if err != nil {
+		return nil, err
+	}
 
-	case params.PMMServerConfig != nil:
-		if err := checkPMMServerLocationConfig(params.PMMServerConfig); err != nil {
-			return nil, err
-		}
-		row.Type = PMMServerBackupLocationType
-		row.PMMServerConfig = params.PMMServerConfig
+	// remove previous configuration
+	row.Type = ""
```

Comment:
so we always expect that we have some config in params even when we want to change only name?

Reply:
Yes, I was following similar funcition in IA channels.

Reply:
Then there is possibility to remove all configs by just sending empty request to this endpoint.
https://github.com/percona/pmm-managed/pull/658/files#diff-72bd90d93d5f3b4bf4a97c3640b8638d111ee250ffd6c77cd54565860912ff5bR161.

And that's where we have a problem in our teams structure and communication: In other change endpoints we change only that fields that we got in request.

Reply:
> In other change endpoints we change only that fields that we got in request. 
So let's keep the same logic in new Change API endpoints to not confuse our customers..